.codegen/model.go.tmpl

```diff
@@ -45,7 +45,7 @@ func (f *{{.PascalName}}) String() string {
 // Set raw string value and validate it against allowed values
 func (f *{{.PascalName}}) Set(v string) error {
     switch v {
-    case {{range $i, $e := .Enum }}{{if $i}}, {{end}}`{{.Content}}`{{end}}:
+    case ``{{range $i, $e := .Enum }}, `{{.Content}}`{{end}}:
```

Comment:
why do we want to allow this?.. can't it be fixed on the platform?..

Reply:
Yeah, I marked it WIP because I'm not sure whether we actually want this or not.

The idea is that optional enum fields are represented as empty strings in structures. When you know the value you want to set an enum field to, you can simply assign it. However, you can't just say `req.EnumField.Set(x)` without checking if `x` is empty. However, an enum may be optional in some contexts and required in others, because optional or not is not a type-level decision, it is a field-level one. So here I propose adding it just in the `Set()` method for all enums to support optional enum fields, basically that a user can use empty string in the `Set()` method to indicate that the field shouldn't be sent in the request without introducing an empty string enum value (which we definitely don't want).